pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -238,5 +269,14 @@ func (o createAuthInfoOptions) validate() error {
 		}
 	}
 
+	if len(o.authProviderArgs) > 0 {
+		newPairs, removePairs, err := cmdutil.ParsePairs(o.authProviderArgs, "auth-provider-arg", true)
```

Comment:
It's reasonable to do this `Complete` so you never have to track `o.authProviderArgs` and can have a non-mutating `Validate` method.  If you're unable to perform the `Complete` step due to bad args, you can fail there.

The `Validate` method allows for clean composition between commands, so that if you built a command on top of this one, you could have the `kubectl fobber` command build a `CreateAuthInfoOptions`, call `Validate` (couldn't call `Complete` since args are likely different), and then have a high confidence that `Run` would work.

@fabianofranz something to watch for so get clean composition out of commands.